tests/runtime/engine/parser.py

```diff
@@ -124,6 +126,8 @@ def __read_test_struct(test, test_suite):
         will_fail = False
         new_pidns = False
         skip_if_env_has = None
+        return_code = 0
+        has_return_code = False
```

Comment:
```suggestion
        return_code = None
```
My Python's rusty, but I think we can do something like this to avoid using two variables when one will do. Then check `return_code is not None` instead of `has_return_code`